src/anonymousUserTracking/criteriaCompletionChecker.ts

```diff
@@ -372,7 +372,10 @@ class CriteriaCompletionChecker {
   }
 
   private getFieldValue(data: any, field: string): any {
-    const fields = field.split('.');
+    let fields = field.split('.');
+    if (data?.eventType === TRACK_EVENT && data?.eventName === fields[0]) {
```

Comment:
Is this relevant only to TRACK_EVENT? Or are there other events that are also affected by this issue?

Reply:
This is for all whose event type is customEvent